datafusion/core/src/datasource/file_format/write/mod.rs

```diff
@@ -150,19 +74,13 @@ pub trait BatchSerializer: Sync + Send {
     fn serialize(&self, batch: RecordBatch, initial: bool) -> Result<Bytes>;
 }
 
-/// Returns an [`AbortableWrite`] which writes to the given object store location
+/// Returns an [`AsyncWrite`] which writes to the given object store location
 /// with the specified compression
 pub(crate) async fn create_writer(
     file_compression_type: FileCompressionType,
     location: &Path,
     object_store: Arc<dyn ObjectStore>,
-) -> Result<AbortableWrite<Box<dyn AsyncWrite + Send + Unpin>>> {
-    let (multipart_id, writer) = object_store
-        .put_multipart(location)
-        .await
-        .map_err(DataFusionError::ObjectStore)?;
-    Ok(AbortableWrite::new(
-        file_compression_type.convert_async_writer(writer)?,
-        MultiPart::new(object_store, multipart_id, location.clone()),
-    ))
+) -> Result<Box<dyn AsyncWrite + Send + Unpin>> {
+    let buf_writer = BufWriter::new(object_store, location.clone());
+    Ok(file_compression_type.convert_async_writer(buf_writer)?)
```

Comment:
I think `create_writer` could still be saved cause it create the `writer` with compression?